### PR TITLE
Fix: end to end test bugs

### PR DESCRIPTION
**Pull request by jniles**
This PR fixes two end to end test bugs:

 1. The end to end tests have been updated for 2021.
 2. The employee config test correctly checks if the `bhCheckboxTree` component is active before blindly clicking on it.

It also includes one cosmetic change to `CHROME_OPTIONS` allowing devs to pass in multiple options by separating them with spaces as so:
```sh
CHROME_OPTIONS="--headless --disable-extensions"
```

Previous PR #5344

### CODE REVIEW

File: protractor.conf.js [hunk 1]
```diff
@@ -80,7 +80,10 @@ if (process.env.CI) {
 
 // Add custom Chrome options for local environments
 if (!process.env.CI && process.env.CHROME_OPTIONS) {
-  config.capabilities.chromeOptions.args.push(process.env.CHROME_OPTIONS);
+
+  // normalize options so that multiple options can be used.
+  const opts = process.env.CHROME_OPTIONS.split(' ');
```
Comment (jmcameron) [inline protractor.conf.js:85]: Nice!

**Review by jmcameron — COMMENTED** [review]
The fixes look okay to me, but I did run into a problem when I tried the end-to-end tests on my laptop:

image

Not sure if this is related or not.  I have not run end-to-end tests on my system for a while so this could be a local issue.

**Review by jmcameron — COMMENTED** [review]
I ran all 4 tests on my laptop. units and integration pass.  E2E failed when trying to create a new FY.  Note that this is further than when I run it directly on the refactor branch or main (on my PC).

**Comment by jniles** [comment]
Can you post the error?  The FY was the explicit fix of this PR.

**Comment by jmcameron** [comment]
`
Fiscal Year
    ✓ blocks invalid form submission with relevant error classes (1541ms)
  errors #interceptor() Error: ER_DUP_ENTRY: Duplicate entry '20210' for key 'period.PRIMARY'
  errors     at Query.Sequence._packetToError (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
  errors     at Query.ErrorPacket (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
  errors     at Protocol._parsePacket (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Protocol.js:291:23)
  errors     at Parser._parsePacket (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Parser.js:433:10)
  errors     at Parser.write (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Parser.js:43:10)
  errors     at Protocol.write (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Protocol.js:38:16)
  errors     at Socket.<anonymous> (/home/jmcameron/Bhima/node_modules/mysql/lib/Connection.js:88:28)
  errors     at Socket.<anonymous> (/home/jmcameron/Bhima/node_modules/mysql/lib/Connection.js:526:10)
  errors     at Socket.emit (events.js:315:20)
  errors     at addChunk (_stream_readable.js:295:12)
  errors     at readableAddChunk (_stream_readable.js:271:9)
  errors     at Socket.Readable.push (_stream_readable.js:212:10)
  errors     at TCP.onStreamRead (internal/stream_base_commons.js:186:23)
  errors     --------------------
  errors     at Protocol._enqueue (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Protocol.js:144:48)
  errors     at PoolConnection.query (/home/jmcameron/Bhima/node_modules/mysql/lib/Connection.js:198:25)
  errors     at queryConnection (/home/jmcameron/Bhima/bin/server/lib/db/transaction.js:14:28)
  errors     at /home/jmcameron/Bhima/bin/server/lib/db/transaction.js:127:48
  errors     at Array.map (<anonymous>)
  errors     at Query.<anonymous> (/home/jmcameron/Bhima/bin/server/lib/db/transaction.js:127:34)
  errors     at Query.<anonymous> (/home/jmcameron/Bhima/node_modules/mysql/lib/Connection.js:526:10)
  errors     at Query._callback (/home/jmcameron/Bhima/node_modules/mysql/lib/Connection.js:488:16)
  errors     at Query.Sequence.end (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/sequences/Sequence.js:83:24)
  errors     at Query._handleFinalResultPacket (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/sequences/Query.js:149:8)
  errors     at Query.OkPacket (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/sequences/Query.js:74:10)
  errors     at Protocol._parsePacket (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Protocol.js:291:23)
  errors     at Parser._parsePacket (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Parser.js:433:10)
  errors     at Parser.write (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Parser.js:43:10)
  errors     at Protocol.write (/home/jmcameron/Bhima/node_modules/mysql/lib/protocol/Protocol.js:38:16)
  errors     at Socket.<anonymous> (/home/jmcameron/Bhima/node_modules/mysql/lib/Connection.js:88:28) {
  errors   code: 'ER_DUP_ENTRY',
  errors   errno: 1062,
  errors   sqlMessage: "Duplicate entry '20210' for key 'period.PRIMARY'",
  errors   sqlState: '23000',
  errors   index: 0,
  errors   sql: "CALL CreateFiscalYear(1, 6, 1, 'A Special Fiscal Year', 12, '2021-12-31 16:00:00.000', '2022-12-30 16:00:00.000', 'Note for the new fiscal Year', @fiscalYearId);"
  errors } +10s
    1) creates a new fiscalYear


  155 passing (4m)
  5 pending
  1 failing

  1) Fiscal Year
       creates a new fiscalYear:

      Expected a success notification, but could not find one.
      + expected - actual

      -false
      +true

      at Object.hasSuccess (/home/jmcameron/Bhima/test/end-to-end/shared/components/notify.js:12:10)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at Context.<anonymous> (/home/jmcameron/Bhima/test/end-to-end/fiscalYears/fiscalYears.spec.js:47:5)
  From: Task: Fiscal Year creates a new fiscalYear
      at Context.ret (/home/jmcameron/Bhima/node_modules/selenium-webdriver/testing/index.js:182:10)
      at processImmediate (internal/timers.js:456:21)

`

**Comment by jniles** [comment]
Thanks!  This looks like a timezone issue that makes it work on GMT+1 and not in Pacific time.  

I will take a look at this further after we merge refactor stock.

**Pull request closed without merging**